images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,329 @@
+
+EXAMPLES = """
+
+- name: Create Kubernetes Event
+  k8s_events
+    state: present
+    name: test-https-emily109
+    namespace: default
+    message: message here
+    reason: reason is now different againnnnnnnnnnnn
+    reportingComponent: reportingComponents here
+    type: Normal
+    source:
+    component: Metering components
+    involvedObject:
+      apiVersion: v1
+      kind: Event
+      name: involvedObject event names
+      namespace: default
+"""
+
+RETURN = """
+result:
+  description:
+  - The created, patched, or otherwise present Event object. Will be empty in the case of a deletion.
+  returned: success
+  type: complex
+  contains:
+     api_version:
+       description: The versioned schema of this representation of an object.
+       returned: success
+       type: str
+     kind:
+       description: Always 'Event'.
+       returned: success
+       type: str
+     metadata:
+       description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+       returned: success
+       type: complex
+     spec:
+       description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+       returned: success
+       type: complex
+     status:
+       description: Current status details for the object.
+       returned: success
+       type: complex
+"""
+
+import copy
+import traceback
+import datetime
+import kubernetes.config.dateutil
+import openshift
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+# user args
+EVENT_ARG_SPEC = {
+    "state": {"default": "present", "choices": ["present", "absent"]},
+    "name": {"required": True},
+    "namespace": {"required": True},
+    "merge_type": {"type": "list", "choices": ["json", "merge", "strategic-merge"]},
+    "message": {"type": "str", "required": True},
+    "reason": {"type": "str", "required": True},
+    "reportingComponent": {"type": "str", "required": True},
+    "type": {"choices": ["Normal", "Warning"]},
+    "source": {"type": "str", "component": {"type": "str", "required": True}},
+    "involvedObject": {
+        "type": "dict",
+        "apiVersion": {"type": "str", "required": True},
+        "kind": {"type": "str", "required": True},
+        "name": {"type": "str", "required": True},
+        "namespace": {"type": "str", "required": True},
+    },
+}
+
+
+class KubernetesEvent(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubernetesEvent, self).__init__(*args, k8s_kind="Event", **kwargs)
+
+    @property
+    def argspec(self):
+        """ argspec property builder """
+        argumentSpec = copy.deepcopy(AUTH_ARG_SPEC)
+        argumentSpec.update(EVENT_ARG_SPEC)
+        return argumentSpec
+
+    def execute_module(self):
+        """ Module execution """
+        #auth
+        self.client = self.get_api_client()
+
+        # retrieve params
```

Comment:
I think it would be cleaner to initialize the constants here:
```python
api_version = 'v1'
event_kind_name = 'Event'
prior_event = None
prior_count = 1
```

What do you think?

Reply:
I'm opened to suggestions on: `event_kind_name`

Reply:
I don't particularly think we even raelly need to use variables for the apiVersion and kind. It doesn't make it any clearer, that's for sure. The amount of repetition is more since the value is shorter than the variable, and not every object shares an apiVersion so the re-use is minimal anyways.

Reply:
Fair enough, but we reference these values multiple times throughout the module, which is why I suggested we declare a variable for it.

Reply:
Going to keep ordering as is. It's clear. Retrieval parameters in one spot.